libopenarc/arc.c

```diff
@@ -3560,7 +3560,57 @@ arc_get_domain(ARC_MESSAGE *msg)
 */
 
 const char *
-arc_chain_str(ARC_MESSAGE *msg)
+arc_chain_status_str(ARC_MESSAGE *msg)
 {
 	return arc_code_to_name(chainstatus, msg->arc_cstate);
 }
+
+/*
+**	ARC_CHAIN_CUSTODY_STR -- retrieve domain chain, as a string
+**
+**	Parameters:
+**      msg -- ARC_MESSAGE object
+**      buf -- where to write
+**      buflen -- bytes at "buf"
+**
+**	Return value:
+**	    Number of bytes written
+*/
+int
+arc_chain_custody_str(ARC_MESSAGE *msg, u_char *buf, size_t buflen)
```

Comment:
Is the right thing to use here a fixed-sized buffer, or a dstring?  As written, this won't overflow the provided buffer, but the caller has no idea if the value got truncated, which could lead to constructing a syntactically invalid A-R or A-A-R field.

Reply:
Working on it.